sklearn/metrics/tests/test_pairwise.py

```diff
@@ -53,34 +55,57 @@
 def test_pairwise_distances():
     # Test the pairwise_distance helper function.
     rng = np.random.RandomState(0)
+
     # Euclidean distance should be equivalent to calling the function.
     X = rng.random_sample((5, 4))
     S = pairwise_distances(X, metric="euclidean")
     S2 = euclidean_distances(X)
     assert_array_almost_equal(S, S2)
+
     # Euclidean distance, with Y != X.
     Y = rng.random_sample((2, 4))
     S = pairwise_distances(X, Y, metric="euclidean")
     S2 = euclidean_distances(X, Y)
     assert_array_almost_equal(S, S2)
+
     # Test with tuples as X and Y
     X_tuples = tuple([tuple([v for v in row]) for row in X])
     Y_tuples = tuple([tuple([v for v in row]) for row in Y])
     S2 = pairwise_distances(X_tuples, Y_tuples, metric="euclidean")
     assert_array_almost_equal(S, S2)
+
+    # Test haversine distance
+    # The data should be valid latitude and longitude
+    X = rng.random_sample((5, 2))
+    X[:, 0] = (X[:, 0] - 0.5) * 2 * np.pi/2
+    X[:, 1] = (X[:, 1] - 0.5) * 2 * np.pi
+    S = pairwise_distances(X, metric="haversine")
+    S2 = haversine_distances(X)
+    assert_array_almost_equal(S, S2)
+
+    # Test haversine distance, with Y != X
+    Y = rng.random_sample((2, 2))
+    Y[:, 0] = (Y[:, 0] - 0.5)*2*np.pi/2
+    Y[:, 1] = (Y[:, 1] - 0.5)*2*np.pi
+    S = pairwise_distances(X, Y, metric="haversine")
+    S2 = haversine_distances(X, Y)
+    assert_array_almost_equal(S, S2)
+
```

Comment:
The code below used to run on a different X

Reply:
Are you sure?
I might be wrong but I cannot find a test with a different X in the git history (the example in the docstring was changed to real places long./lat. but not this example).